experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -164,14 +198,25 @@ export interface MapNode<TSchema extends MapSchema> extends UntypedTree {
 }
 
 /**
- * A node that behaves like a `{ content: Field }` for a specific `Field` type.
+ * A {@link TreeNode} that wraps a single field.
  *
  * @remarks
  * FieldNodes unbox to their content, so in schema aware APIs which do unboxing, the FieldNode itself will be skipped over.
+ * Mainly useful when a field (such as a sequence) is desired in a location where nodes are required (like the member of a union or the child of another field).
+ *
+ * For example:
+ * `Sequence<Foo> | Sequence<Bar>` or `OptionalField<Sequence<Foo>>` can't be expressed as simple fields
+ * (unlike `Sequence<Foo | Bar>` or `OptionalField<Foo>` which can be done as simple fields).
+ * Instead {@link FieldNode}s can be use to achieve something similar, more like:
+ * `FieldNode<Sequence<Foo>> | FieldNode<Sequence<Bar>>` or `OptionalField<FieldNode<Sequence<Foo>>>`.
+ *
+ * This extra layer of field nodes is then omitted when using schema-aware APIs which  do unboxing.
```

Comment:
```suggestion
 * This extra layer of field nodes is then omitted when using schema-aware APIs which do unboxing.
```
Great explanatory example, by the way.